File: 643-maximum-average-subarray-i/maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double sum = 0;
        int n = nums.size();
        for(int i=0; i<k; i++){
            sum += nums[i];
        }
        double maxsum = sum;
        for(int i=k; i<n; i++){
            sum += nums[i] - nums[i - k];
            maxsum = max(maxsum, sum);
        }
        return maxsum / k;
    }
};

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        sum = 0
        for i in range(k):
            sum += nums[i]
        maxsum = sum 
        for i in range(k, len(nums)):
            sum += nums[i] - nums[i - k]
            maxsum = max(maxsum, sum)
        return maxsum / k
        